gt/lua_lpm.c

```diff
@@ -0,0 +1,395 @@
+			lua_gettop(l));
+
+	ret = rte_lpm6_add(*lpm6, ipv6_addr->s6_addr, depth, label);
+	if (ret < 0) {
+		luaL_error(l, "lpm6: failed to add a network policy to the lpm6 table at %s!",
+			__func__);
+	}
+
+	lua_pushinteger(l, 0);
+
+	return 1;
+}
+
+static int
+l_lpm6_del(lua_State *l)
+{
+	int ret;
+	uint8_t depth;
+	struct in6_addr *ipv6_addr;
+	uint32_t ctypeid;
+	uint32_t correct_ctypeid_in6_addr;
+	void *cdata;
+
+	/* First argument must be of type struct rte_lpm6 **. */
+	struct rte_lpm6 **lpm6 =
+		(struct rte_lpm6 **)luaL_checkudata(l, 1, "gt_lpm6");
+
+	correct_ctypeid_in6_addr = luaL_get_ctypeid(l,
+		CTYPE_STRUCT_IN6_ADDR_REF);
+	/* Second argument must be of type CTYPE_STRUCT_IN6_ADDR_REF. */
+	cdata = luaL_checkcdata(l, 2, &ctypeid, CTYPE_STRUCT_IN6_ADDR_REF);
+	if (ctypeid != correct_ctypeid_in6_addr)
+		luaL_error(l, "Expected `%s' as second argument",
+			CTYPE_STRUCT_IN6_ADDR_REF);
+
+	ipv6_addr = (struct in6_addr *)cdata;
+
+	/* Third argument must be a Lua number. */
+	depth = luaL_checknumber(l, 3);
+
+	if (lua_gettop(l) != 3)
+		luaL_error(l, "Expected three arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	ret = rte_lpm6_delete(*lpm6, ipv6_addr->s6_addr, depth);
+
+	lua_pushinteger(l, ret);
+
+	return 1;
+}
+
+static int
+l_lpm6_lookup(lua_State *l)
+{
+	int label;
+	struct in6_addr *ipv6_addr;
+	uint32_t ctypeid;
+	uint32_t correct_ctypeid_in6_addr;
+	void *cdata;
+
+	/* First argument must be of type struct rte_lpm6 **. */
+	struct rte_lpm6 **lpm6 =
+		(struct rte_lpm6 **)luaL_checkudata(l, 1, "gt_lpm6");
+
+	correct_ctypeid_in6_addr = luaL_get_ctypeid(l,
+		CTYPE_STRUCT_IN6_ADDR_REF);
+	/* Second argument must be of type CTYPE_STRUCT_IN6_ADDR_REF. */
+	cdata = luaL_checkcdata(l, 2, &ctypeid, CTYPE_STRUCT_IN6_ADDR_REF);
+	if (ctypeid != correct_ctypeid_in6_addr)
+		luaL_error(l, "Expected `%s' as second argument",
+			CTYPE_STRUCT_IN6_ADDR_REF);
+
+	ipv6_addr = (struct in6_addr *)cdata;
+
+	if (lua_gettop(l) != 2)
+		luaL_error(l, "Expected two arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	label = lpm_lookup_ipv6(*lpm6, ipv6_addr);
+
+	lua_pushinteger(l, label);
+
+	return 1;
+}
+
+static const struct luaL_reg lpmlib_lua_c_funcs [] = {
+	{"str_to_prefix",  l_str_to_prefix},
+	{"new_lpm",        l_new_lpm},
+	{"lpm_add",        l_lpm_add},
+	{"lpm_del",        l_lpm_del},
+	{"lpm_lookup",     l_lpm_lookup},
+	{"str_to_prefix6", l_str_to_prefix6},
+	{"new_lpm6",       l_new_lpm6},
+	{"lpm6_add",       l_lpm6_add},
+	{"lpm6_del",       l_lpm6_del},
+	{"lpm6_lookup",    l_lpm6_lookup},
+	{NULL,             NULL}	/* Sentinel. */
+};
+
+static int
+lpm_gc(lua_State *l) {
+	struct rte_lpm *lpm = *(struct rte_lpm **)lua_touserdata(l, 1);
+	rte_lpm_free(lpm);
+	return 0;
+}
+
+static int
+lpm6_gc(lua_State *l) {
+	struct rte_lpm6 *lpm6 = *(struct rte_lpm6 **)lua_touserdata(l, 1);
+	rte_lpm6_free(lpm6);
+	return 0;
+}
+
+void
+lualpm_openlib(lua_State *l) {
+	luaL_newmetatable(l, "gt_lpm");
```

Comment:
Create a `#define` for `"gt_lpm"` and avoid all the duplicates in the code.

Do the same for `"gt_lpm6"`.